### PR TITLE
Add red border when required TextField is empty and goes out of focus

### LINKED ISSUE #8213: No red border when a text field is invalid
# Environment Information 
 
- __Package version(s)__: `@microsoft/sp-office-ui-fabric-core` = 1.7.1
- __Browser and OS versions__: Chrome on Windows 10

# Actual behavior

I've a required text field and a required date picker. See code below:

```jsx
<TextField
    label="Text"
    required
    value=""
/>
<DatePicker
    isRequired
    label="Date"
    value=""
/>
```

Both got a difference way to show that there is an error on that field when the field has been unfocused: The text field don't has a red border in contract to the date picker that has a red border when it has been unfocused. [See this code pen](https://codepen.io/HPawelyn/pen/RdKVbe?editors=0010)

# Expected behavior

The same behavior for the text fields like the date pickers: A red border when the text field is invalid when unfocused.

# Priorities and help requested:
 
- **Are you willing to submit a PR to fix?** No
- **Requested priority:** Low

### PR DESCRIPTION
**Pull request by aftab-hassan**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #8213 
- [x] Include a change request file using `$ npm run change`

#### Description of changes
If it is a `required` field, set the color of the `TextField` to `red` on losing `focus`, and unset the `red` color when `focus` is regained. 

#### Focus areas to test
1. Have a required TextField
2. Place focus on the TextField
3. Click elsewhere leaving the TextField empty
4. The border of the TextField should now be `red`

red-border

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8311)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx [hunk 1]
```diff
@@ -230,7 +234,7 @@ export class TextFieldBase extends BaseComponent<ITextFieldProps, ITextFieldStat
       <div className={this._classNames.root}>
         <div className={this._classNames.wrapper}>
           {onRenderLabel(this.props, this._onRenderLabel)}
-          <div className={this._classNames.fieldGroup}>
+          <div className={this._classNames.fieldGroup} style={{ border: this.state.color }}>
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx]: We should not use the style tag for that when we already have the className attached to this element. Better to have some kind of required boolean flag as a style prop passed to getStyles and do the magic using the appropriate palette or semantic colors.

File: packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx [hunk 2]
```diff
@@ -441,6 +456,7 @@ export class TextFieldBase extends BaseComponent<ITextFieldProps, ITextFieldStat
         aria-invalid={!!this.state.errorMessage}
         aria-label={this.props.ariaLabel}
         readOnly={this.props.readOnly}
+        required={this.props.required}
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx]: Also, the `required` native prop will work only if the TextField is placed within a `form` and clicking the submit button. Good to have it though... unless the `getNativeProps` is already setting it at top and you are just reassigning it here.

File: packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx (hunk 1 above)
Comment (aftab-hassan) [reply packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx]: Thanks @Vitalius1 

Shall incorporate the suggestions once we decide on the behavior we'd like to use. Until then, I'll keep the `PR` closed.

**Comment by aftab-hassan** [comment]
Temporarily closing this PR until we decide on the default behavior to go with. Discussion here: https://github.com/OfficeDev/office-ui-fabric-react/issues/8213

**Pull request closed without merging**